src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -4102,40 +4130,56 @@ WorkerLimitCount(MultiExtendedOp *originalOpNode)
 	 * aggregates, we can push down the original limit. Else if we have order by
 	 * clauses with commutative aggregates, we can push down approximate limits.
 	 */
-	if (groupClauseList == NIL)
+	if (groupClauseList == NIL || sortClauseList == NIL || !hasOrderByAggregate)
```

Comment:
I don't know if we changed the code here or I missed it before, but anyway, I think we have a problem here.

If we run this query on the master, it will push down limit to worker nodes, and this would cause wrong results.

I think that previously we didn't push down limit in this case.

select l_orderkey, count(*) from lineitem group by l_orderkey limit 10;

A set of results would be;
postgres=# select l_orderkey, count(*) from lineitem group by l_orderkey limit 10;
 l_orderkey | count
------------+-------
        129 |     7
       2848 |     5
       2054 |     7
       3623 |     7
        195 |     4
       1509 |     7
        705 |     2
       2918 |     1
        480 |     1
       1126 |     3
(10 rows)


But there would some "129" on different worker nodes which are left outside due to pushed down limit.